### PR TITLE
yarn: init at 0.17.8

### PR DESCRIPTION
**Pull request by moretea**
###### Motivation for this change
Yarn is a new javascript packagemanager by Facebook, inspired by Bundler and Cargo.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/node-packages/node-env.nix [hunk 1]
```diff
@@ -112,47 +112,40 @@ let
...
+          
           # Restore write permissions to make building work
           find "$packageDir" -type d -print0 | xargs -0 chmod u+x
           chmod -R u+w "$packageDir"
-
+          
           # Move the extracted tarball into the output folder
           mv "$packageDir" "$DIR/${packageName}"
       elif [ -d "${src}" ]
       then
```
Comment (joachifm) [inline pkgs/development/node-packages/node-env.nix]: The comment says that `strippedName` would be set on older nixpkgs, implying that it is *unset* in newer versions (and so must be set). Is that not the case?

**Comment by joachifm** [comment]
cc @svanderburg might be able to comment on this.

File: pkgs/development/node-packages/node-env.nix (hunk 1 above)
Comment (moretea) [inline pkgs/development/node-packages/node-env.nix]: I only added one line in  `pkgs/top-level/node-packages.json`, the other changes are auto generated by a script, invoked to the letter of the instructions AFAIK.

**Comment by globin** [comment]
Tested and everything seems to work.

**Comment by moretea** [comment]
Then I'll update it :)

**Pull request merged**